sql/core/src/test/scala/org/apache/spark/sql/StreamTest.scala

```diff
@@ -179,16 +206,38 @@ trait StreamTest extends QueryTest with Timeouts {
          |${if (streamDeathCause != null) stackTraceToString(streamDeathCause) else ""}
          |
          |== Sink ==
-         |$sink
+         |${sink.toLongString}
          |
          |== Plan ==
          |${if (currentStream != null) currentStream.lastExecution else ""}
-         """
+         """.stripMargin
+
+    def assert(condition: => Boolean, message: String): Unit = {
+      try {
+        Assertions.assert(condition)
+      } catch {
+        case NonFatal(e) =>
+          failTest(message, e)
+      }
+    }
 
-    def checkState(check: Boolean, error: String) = if (!check) {
+    def eventually[T](message: String)(func: => T): T = {
+      try {
+        Eventually.eventually(Timeout(streamingTimout)) {
+          func
+        }
+      } catch {
+        case NonFatal(e) =>
+          failTest(message, e)
+      }
+    }
+
+    def failTest(message: String, cause: Throwable = null) = {
+      val c = Option(cause).map { _.getStackTrace.take(10).mkString("", "\n|\t", "\n") }
```

Comment:
convenience method to remove a lot of duplicated code to generate testState string.